packages/rlc-common/src/buildSamples.ts

```diff
@@ -0,0 +1,835 @@
+
+  requestParameter.parameters
+    ?.filter((p) => p.type === "query")
+    .forEach((p) => {
+      const name = `${p.name}`;
+      querySideAssignments.push(
+        `${name}: ` + mockParameterTypeValue(p.param.type, name, schemaMap)
+      );
+    });
+
+  if (querySideAssignments.length > 0) {
+    allSideAssignments.push(
+      ` queryParameters: { ` + querySideAssignments.join(", ") + `}`
+    );
+  }
+  requestParameter.parameters
+    ?.filter((p) => p.type === "header")
+    .filter((p) => p.name.toLowerCase() !== "contenttype")
+    .forEach((p) => {
+      const name = `${p.name}`;
+      headerSideAssignments.push(
+        `${name}: ` + mockParameterTypeValue(p.param.type, name, schemaMap)
+      );
+    });
+  if (headerSideAssignments.length > 0) {
+    allSideAssignments.push(
+      ` headers: { ` + headerSideAssignments.join(", ") + `}`
+    );
+  }
+  const contentType = requestParameter.parameters
+    ?.filter((p) => p.type === "header")
+    .filter((p) => p.name.toLowerCase() === "contenttype");
+  if (contentType && contentType.length > 0) {
+    allSideAssignments.push(
+      ` ${contentType[0].name}: ` +
+        mockParameterTypeValue(
+          contentType[0].param.type,
+          contentType[0].name,
+          schemaMap
+        )
+    );
+  }
+  let value: string = `{}`;
+  if (allSideAssignments.length > 0) {
+    value = `{ ` + allSideAssignments.join(", ") + `}`;
+  } else {
+    return [];
+  }
+
+  const optionParam: SampleParameter = {
+    name: "options",
+    assignment: `const options: ${method.optionsName} =` + value + `;`
+  };
+  addValueInImportedDict(packageName, method.optionsName, importedDict);
+  return [optionParam];
+}
+
+function addValueInImportedDict(
+  key: string,
+  val: string,
+  importedDict: Record<string, Set<string>>
+) {
+  if (!importedDict[key]) {
+    importedDict[key] = new Set<string>();
+  }
+  importedDict[key].add(val);
+}
+
+function buildMethodParamMap(model: RLCModel): Map<string, OperationParameter> {
+  const map = new Map<string, OperationParameter>();
+  (model.parameters ?? []).forEach((p) => {
+    const operatonConcante = getOperationConcate(
+      p.operationName,
+      p.operationGroup,
+      model.options?.sourceFrom
+    );
+    map.set(operatonConcante, p);
+  });
+  return map;
+}
+
+function getOperationConcate(
+  opName: string,
+  opGroup: string,
+  sourceFrom?: string
+) {
+  return sourceFrom === "Swagger"
+    ? opGroup === "" || opGroup === "Client"
+      ? opName
+      : `${opGroup}${opName}`
+    : `${opGroup}_${opName}`;
+}
+
+function buildSchemaObjectMap(model: RLCModel) {
+  // include interfaces
+  const map = new Map<string, Schema>();
+  const allSchemas = (model.schemas ?? []).filter(
+    (o) =>
+      isObjectSchema(o) &&
+      (o as ObjectSchema).usage?.some((u) => [SchemaContext.Input].includes(u))
+  );
+  allSchemas.forEach((o) => {
+    map.set(o.name, o);
+  });
+
+  return map;
+}
+
+function mockParameterTypeValue(
```

Comment:
Can you put all methods for creating mock values in a separate file?  This could help to reduce the file size and simply the core logic.